vfs_test.go

```diff
@@ -174,3 +164,108 @@ func TestVFSFH(t *testing.T) {
 	require.NoError(t, r.Close())
 	assert.ElementsMatch(t, b, bRead)
 }
+
+// TestVFSLs validates vfs LsDir operation is successful
+func TestVFSLs(t *testing.T) {
+	config, err := NewConfig()
+	require.NoError(t, err)
+
+	context, err := NewContext(config)
+	require.NoError(t, err)
+
+	vfs, err := NewVFS(context, config)
+	require.NoError(t, err)
+
+	tmpPath := filepath.Join(t.TempDir(), "somedir")
+	tmpPath2 := filepath.Join(tmpPath, "somedir2")
+	tmpPath3 := filepath.Join(tmpPath, "somedir3")
+
+	isDir, err := vfs.IsDir(tmpPath)
+	require.NoError(t, err)
+	assert.False(t, isDir)
+
+	isDir, err = vfs.IsDir(tmpPath3)
+	require.NoError(t, err)
+	assert.False(t, isDir)
+
+	isDir, err = vfs.IsDir(tmpPath3)
+	require.NoError(t, err)
+	assert.False(t, isDir)
+
+	tmpFilePath := filepath.Join(tmpPath, "somefile")
+	tmpFilePath2 := filepath.Join(tmpPath, "somefile2")
+	tmpFilePath3 := filepath.Join(tmpPath, "somefile3")
+
+	isFile, err := vfs.IsFile(tmpFilePath)
+	require.NoError(t, err)
+	assert.False(t, isFile)
+
+	isFile, err = vfs.IsFile(tmpFilePath2)
+	require.NoError(t, err)
+	assert.False(t, isFile)
+
+	isFile, err = vfs.IsFile(tmpFilePath3)
+	require.NoError(t, err)
+	assert.False(t, isFile)
+
+	// Create directories
+	require.NoError(t, vfs.CreateDir(tmpPath))
+	require.NoError(t, vfs.CreateDir(tmpPath2))
+	require.NoError(t, vfs.CreateDir(tmpPath3))
+
+	isDir, err = vfs.IsDir(tmpPath)
+	require.NoError(t, err)
+	assert.True(t, isDir)
+
+	isDir, err = vfs.IsDir(tmpPath2)
+	require.NoError(t, err)
+	assert.True(t, isDir)
+
+	isDir, err = vfs.IsDir(tmpPath3)
+	require.NoError(t, err)
+	assert.True(t, isDir)
+
+	// Create Files
+	createFile(t, vfs, tmpFilePath)
+	createFile(t, vfs, tmpFilePath2)
+	createFile(t, vfs, tmpFilePath3)
+
+	isFile, err = vfs.IsFile(tmpFilePath)
+	require.NoError(t, err)
+	assert.True(t, isFile)
+
+	isFile, err = vfs.IsFile(tmpFilePath2)
+	require.NoError(t, err)
+	assert.True(t, isFile)
+
+	isFile, err = vfs.IsFile(tmpFilePath3)
+	require.NoError(t, err)
+	assert.True(t, isFile)
+
+	folderList, fileList, err := vfs.Ls(tmpPath)
+	require.NoError(t, err)
+	assert.Equal(t, 2, len(folderList))
+	assert.Equal(t, 3, len(fileList))
+
+	// Remove Files
+	require.NoError(t, vfs.RemoveFile(tmpFilePath))
```

Comment:
Since we're not testing vfs.RemoveFile I would just let the `t.TempDir` cleanup handle deleting all these files itself.

Reply:
👍